### ISSUE TITLE
EmbeddingBag documentation mistake

### ISSUE THREAD

**Issue by ZeliTheZealot** [pytorch/pytorch#43197]
## 📚 Documentation

In [`EmbeddingBag`](https://pytorch.org/docs/stable/generated/torch.nn.Embedding.html#torch.nn.Embedding), it is stated: 

with `mode="mean"` is equivalent to `Embedding` followed by `torch.mean(dim=0)`

However, I think the author intends `dim=1` instead. This is seen by comparing the following example ... 
```
>>> # an Embedding module containing 10 tensors of size 3
>>> embedding_sum = nn.EmbeddingBag(10, 3, mode='sum')
>>> # a batch of 2 samples of 4 indices each
>>> input = torch.LongTensor([1,2,4,5,4,3,2,9])
>>> offsets = torch.LongTensor([0,4])
>>> embedding_sum(input, offsets)
tensor([[-0.8861, -5.4350, -0.0523],
        [ 1.1306, -2.5798, -1.0044]])
```
... with the corresponding example in [`Embedding`](https://pytorch.org/docs/stable/generated/torch.mean.html).

```
>>> # an Embedding module containing 10 tensors of size 3
>>> embedding = nn.Embedding(10, 3)
>>> # a batch of 2 samples of 4 indices each
>>> input = torch.LongTensor([[1,2,4,5],[4,3,2,9]])
>>> embedding(input)
tensor([[[-0.0251, -1.6902,  0.7172],
         [-0.6431,  0.0748,  0.6969],
         [ 1.4970,  1.3448, -0.9685],
         [-0.3677, -2.7265, -0.1685]],

        [[ 1.4970,  1.3448, -0.9685],
         [ 0.4362, -0.4004,  0.9400],
         [-0.6431,  0.0748,  0.6969],
         [ 0.9124, -2.3616,  1.1151]]])
```

cc @jlin27

**Comment by SplitInfinity** [comment]
I think documentation is correct, but maybe could be clearer. The reduction with `dim=0` is logically applied to each *bag*, not to the output tensor from the `Embedding` that `EmbeddingBag` implicitly computes:

> **mode** (*string*, *optional*) – `"sum", "mean" or "max"`. Specifies the way to reduce the bag. `"sum"` computes the weighted sum, taking per_sample_weights into consideration. "mean" computes the average of the values in the bag, `"max"` computes the max value over each bag. Default:` "mean"`

Each bag is reduced with `dim=0`. What is considered to be a bag depends on the inputs to `EmbeddingBag`:

> - If input is 2D of shape `(B, N)`, it will be treated as `B` bags (sequences) each of fixed length `N`, and this will return `B` values aggregated in a way depending on the `mode`. `offsets` is ignored and required to be `None` in this case.

```
>>> weights = torch.nn.Parameter(torch.randn(10, 3))                 
>>> e = torch.nn.Embedding(10, 3)                  
>>> eb = torch.nn.EmbeddingBag(10, 3, mode="sum")              
>>> e.weight = weights                             
>>> eb.weight = weights                           
>>> input = torch.LongTensor([[1,2,4,5],[4,3,2,9]])
# Since input is 2D, we have 2 bags of 4 indices and dim=1 for sum because that is dim=0 for each bag
>>> torch.sum(e(input), dim=1)                             
tensor([[-3.6195,  4.2928, -0.8031],
        [-3.2539,  2.0035,  0.7717]], grad_fn=<SumBackward1>)
>>> eb(input)
tensor([[-3.6195,  4.2928, -0.8031],
        [-3.2539,  2.0035,  0.7717]], grad_fn=<EmbeddingBagBackward>)
```

> - If input is 1D of shape `(N)`, it will be treated as a concatenation of multiple bags (sequences). `offsets` is required to be a 1D tensor containing the starting index positions of each bag in input. Therefore, for offsets of shape `(B)`, input will be viewed as having `B` bags. Empty bags (i.e., having 0-length) will have returned vectors filled by zeros.

```
>>> weights = torch.nn.Parameter(torch.randn(10, 3))                 
>>> e = torch.nn.Embedding(10, 3)                  
>>> eb = torch.nn.EmbeddingBag(10, 3, mode="sum")              
>>> e.weight = weights                             
>>> eb.weight = weights                           
>>> input = torch.LongTensor([1,2,4,5,4,3,2,9])
>>> offsets = torch.LongTensor([0,4])
# It's a lot of code, but this line indexes into the output of `Embedding` to gather indices 0-3 into one tensor (bag 0), 4-8 into another (bag 1), sums those using dim=0 because the reduction is applied to bags, and concatenates the results
>>> torch.stack([torch.sum(e(input)[offsets[i]:(offsets[i+1] if i != len(offsets)-1 else e(input).shape[0])], dim=0) for i in range(len(offsets))])                          
tensor([[-3.6195,  4.2928, -0.8031],
        [-3.2539,  2.0035,  0.7717]], grad_fn=<SumBackward1>)
>>> eb(input, offsets)
tensor([[-3.6195,  4.2928, -0.8031],
        [-3.2539,  2.0035,  0.7717]], grad_fn=<EmbeddingBagBackward>)
```
